File: exlang/exp/two_sum.py
def two_sum(nums, target):
    # Dictionary to store number and its index
    num_to_index = {}
    
    # Iterate over the list with index and value
    for i, num in enumerate(nums):
        # Compute the complement needed to reach the target
        complement = target - num
        
        # If the complement is already in the dictionary, return the pair of indices
        if complement in num_to_index:
            return [num_to_index[complement], i]
        
        # Otherwise, store the current number with its index
        num_to_index[num] = i

    # If no solution is found (shouldn't happen as per problem constraints)
    return []

# Example usage:
if __name__ == '__main__':
    nums = [2, 7, 11, 15]
    target = 9
    result = two_sum(nums, target)
    print(result)  # Expected output: [0, 1]

File: exapi/exj/two-sum.js
function twoSum(nums, target) {
  // Create a hash map to store each number and its index
  const map = new Map();
  
  // Loop through the array
  for (let i = 0; i < nums.length; i++) {
    // Determine the number needed to reach the target
    const complement = target - nums[i];
    
    // If the complement exists in the map, return the indices
    if (map.has(complement)) {
      return [map.get(complement), i];
    }
    
    // Otherwise, store the current number and its index in the map
    map.set(nums[i], i);
  }
  
  // If no solution is found, which should not happen as the problem guarantees one solution
  return [];
}

// Example usage:
const nums = [2, 7, 11, 15];
const target = 9;
console.log(twoSum(nums, target)); // Expected output: [0, 1]
